src/imperator/data/setup_main.rs

```diff
@@ -0,0 +1,295 @@
+            validate_great_works(block, data);
+        });
+    }
+}
+
+impl FileHandler<Block> for SetupMain {
+    // TODO - Is this impl needed? Copied it from src/vic3/data/history.rs
+    fn subpath(&self) -> PathBuf {
+        PathBuf::from("setup/main/")
+    }
+
+    fn load_file(&self, entry: &FileEntry) -> Option<Block> {
+        if !entry.filename().to_string_lossy().ends_with(".txt") {
+            return None;
+        }
+
+        PdxFile::read(entry)
+    }
+
+    fn handle_file(&mut self, _entry: &FileEntry, mut block: Block) {
+        for (key, block) in block.drain_definitions_warn() {
+            self.load_item(key, block);
+        }
+    }
+}
+
+fn validate_treasures(block: &Block, data: &Everything) {
+    let mut vd = Validator::new(block, data);
+    vd.field_validated_block("database", |block, data| {
+        let mut vd = Validator::new(block, data);
+        for (_, block) in vd.integer_blocks() {
+            let mut vd = Validator::new(block, data);
+            vd.field_item("key", Item::Localization);
+            vd.choice(DLC_IMPERATOR);
+            vd.field("icon"); // TODO - icon can be any icon declared in "gfx/interface/icons/treasures", how to check that?
+            vd.multi_field_validated_block("state_modifier", |block, data| {
+                validate_modifs(block, data, ModifKinds::Country | ModifKinds::Province | ModifKinds::State, vd);
+            });
+        }
+    });
+}
+
+fn validate_families(block: &Block, data: &Everything) {
+    let mut vd = Validator::new(block, data);
+    vd.field_validated_block("families", |block, data| {
+        let mut vd = Validator::new(block, data);
+        for (_, block) in vd.integer_blocks() {
+            let mut vd = Validator::new(block, data);
+            vd.field_item("key", Item::Localization);
+            vd.field_item("owner", Item::Country); // can be any country tag declared in setup
+            vd.field_item("culture", Item::Country);
+            vd.field_integer("prestige");
+            vd.field_integer("color");
+        }
+    });
+}
+fn validate_diplomacy(block: &Block, data: &Everything) {
+    let mut vd = Validator::new(block, data);
+    vd.multi_field_validated_block("defensive_league", |block, data| {
+        let mut vd = Validator::new(block, data);
+        vd.multi_field_item("member", Item::Country);
+    });
+    for field in &["dependency", "dependency", "guarantee", "alliance"] {
+        vd.multi_field_validated_block(field, |block, data| {
+            let mut vd = Validator::new(block, data);
+            vd.field_item("first", Item::Country);
+            vd.field_item("second", Item::Country);
+            if field == "dependency" {
+                vd.field_item("subject_type", Item::SubjectType);
+            }
+        });
+    }
+    /* 
+    TODO - imperator - This block also has a trade_access section with irregular syntax, not sure how to do this, it has two nested Item::Country basically
```

Comment:
You could use the unknown_block_field approach (honestly I don't remember if that's the right method name) and check that the key is a country, then create an inner validator and do the same to that one .